### ISSUE TITLE
Another license option for gem creation?

### ISSUE THREAD

**Issue by CoralineAda** [ruby/rubygems#4087]
### Describe the problem as clearly as you can

bundler only seems to support the MIT license.

### Post steps to reproduce the problem

`bundle gem foo`

From the docs: 

"LICENSE.txt: Includes the MIT license. Will only be included if you chose to have it included."

### Question for the community

Would you accept a PR that added the [Hippocratic License 2.1](https://firstdonoharm.dev/) as a licensing option?

**Comment by deivid-rodriguez** [comment]
I would personally be happy with adding this.

I wonder, however, whether we should go all the way and provide a `--license` option that accepts any [spdx license identifier](https://spdx.org/licenses/).

**Comment by colby-swandale** [comment]
I agree, I think adding more licenses and providing an option to choose & set a default is a great step forward.

**Comment by hsbt** [comment]
I also agree to add the selectable options with SPDX identifier like `bundle gem --license=BSD-2-Clause`. But I'm against to add the specific option for Hippocratic License.

**Comment by CoralineAda** [comment]
Bundler normalized the MIT license. Bundler normalized codes of conduct. Bundler can normalize ethical licenses.

**Comment by deivid-rodriguez** [comment]
@CoralineAda To make sure I got you correctly, are you ok with the proposed solution? Or did you have something else in mind when you offered a PR?

**Comment by CoralineAda** [comment]
My suggestion was to simply add the HL as an alternative to the default MIT license, to encourage and support the Ruby community in adopting ethical licensing.

**Comment by simi** [comment]
I'm also against adding additional special licence. Making it more general would be more useful as mentioned by @hsbt.

> Bundler helped normalize the MIT license. Bundler helped normalize codes of conduct. Bundler can help normalize ethical licenses.

Bundler can help normalize custom license selection actually. That would be the best IMHO.

**Comment by deivid-rodriguez** [comment]
The way I see it, if we add a special CLI flag for the hippocratic license in order to promote ethical licensing, then I believe people will start requesting their favourite open source license to be added too. And when that happens, we can do two things:

* Reject those, since as more flags are added, the ethical license would start getting more hidden between other flags to the point where it wouldn't be promoting it in a special way anymore.

* Accept those, then we would end up not promoting the license in a special way and we would end up with a mess of CLI flags that we would like to refactor into a generic `--license` flag.

None of those options seem particularly appealing, so my proposal would be to add support for the hippocratic license through a generic CLI flag, like `--license  Hippocratic-2.1`, and then promote ethical licenses and encourage adoption through other means, like documentation, blog posts, and so on.

**Comment by deivid-rodriguez** [comment]
Another way to promote it would be to mention Hippocractic license explicit in the CLI help, like:

```

OPTIONS

(...)

       --license=<license> Add a LICENSE to the project. Value must be a spdx license identifier, like "Hippocractic 2.1" or "MIT". Your name from the global git config is used for the copyright statement. If this option is unspecified, an interactive prompt will be displayed and the answer will be saved in Bundler´s global config for future bundle gem use.

(...)
```

and the prompt, like:

```
Which license you would like to use? You can use any valid spdx identifier (https://spdx.org/licenses/) here. For example,
* You can use a copyleft license, such as "Hippocractic 2.1", that specifically prohibits the use of software to violate universal standards of human rights, and embodying the principles of Ethical Source Software. You can read more about the hippocratic license at https://firstdonoharm.dev/.
* You can use a permissive license, like "MIT", that means that any other developer or company will be legally allowed to use your code for free as long as they admit you created it. You can read more about the MIT license at https://choosealicense.com/licenses/mit.
Specify an spdx license identifier: 
```

**Comment by deivid-rodriguez** [comment]
I think my suggestion would be more powerful than an explicit CLI flag, since the text would be read by all first time users of the `--license` flag whereas an explicitly flag would require you to find the option an use it first.

**Comment by deivid-rodriguez** [comment]
Sorry, I don't know a lot about licensing so my wording above is completely incorrect since the hippocratic license is not a copyleft license in the FSF sense, since it restricts usage. We could still recommend it as our preferred non-free license :)

**Comment by CoralineAda** [comment]
@deivid-rodriguez Should I open a PR with your implementation suggestions?

**Comment by deivid-rodriguez** [comment]
Hei @CoralineAda.

I think there's consensus on adding a generic `--license` flag, so we'd definitely accept a PR adding that. Regarding my more recent comment about recommending the hippocratic license in help and CLI messages, I think that requires more consensus and discussion at this point.
